AdyenNetworking/APIClient/HTTPResponse.swift

```diff
@@ -1,12 +1,28 @@
 //
-// Copyright (c) 2021 Adyen N.V.
+// Copyright (c) 2023 Adyen N.V.
 //
 // This file is open source and available under the MIT license. See the LICENSE file for more info.
 //
 
 import Foundation
 
-public struct HTTPResponse<R: Response> {
+/// Protocol representing a HTTP response, containing headers, status code and body
+public protocol AnyHTTPResponse {
```

Comment:
whats the purpose of this protocol if i may ask :-)? it can't be used as a parameter type or a property type, it can only serve as generic constraint, right? do you need that?